### PR TITLE
week2 server side. Added form post handler. Fixed week1 html issues

### PR DESCRIPTION
**Pull request by nourishnew**

### CODE REVIEW

File: portfolio/src/main/webapp/style.css [hunk 1]
```diff
@@ -16,6 +19,34 @@
   flex-direction: row;
   justify-content: space-between;
 }
+.flex-col {
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.1.4](https://google.github.io/styleguide/htmlcssguide.html#ID_and_Class_Name_Delimiters): Separate words in class names by a hyphen.
```suggestion
.flex-col {
```

Here and elsewhere.

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -16,6 +19,34 @@
...
+  display: flex;
+  flex-direction: column;
+}
+form{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.7](https://google.github.io/styleguide/htmlcssguide.html#Rule_Separation): Separate rules by new lines.

```suggestion
}

form {
```

Here and elsewhere.

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -6,6 +6,9 @@
   display: flex;
   flex-direction: row;
 }
+.prim_color{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.1.4](https://google.github.io/styleguide/htmlcssguide.html#ID_and_Class_Name_Delimiters): Separate words in class names by a hyphen.
```suggestion
.prim-color {
```

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -6,6 +6,9 @@
...
+  color:#fca61f;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/style.css [hunk 5]
```diff
@@ -16,6 +19,34 @@
   flex-direction: row;
   justify-content: space-between;
 }
+.flex-col {
+  display: flex;
+  flex-direction: column;
+}
+form{
+  margin:auto;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -16,6 +19,34 @@
...
+  max-width: 70em;
+}
+textarea{
+  border: 1px solid antiquewhite;
+  border-radius: 10px;
+  resize:none;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -16,6 +19,34 @@
...
+  padding:10px;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/style.css [hunk 8]
```diff
@@ -16,6 +19,34 @@
...
+  font-family: 'Nunito Sans', sans-serif;
+  margin:10px;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/style.css [hunk 9]
```diff
@@ -70,6 +101,23 @@ p {
...
+.box-shadow{
+  border-radius: 10px;
+  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
+}
+.margin-div{
+  padding: 5em;
+  margin: 5em;
+}
+.margin-left{
+  margin-left:10px;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/style.css [hunk 10]
```diff
@@ -70,6 +101,23 @@ p {
...
+}
+.margin-top{
+  margin-top:10px;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css]: [4.2.4](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space after a property name’s colon.

File: portfolio/src/main/webapp/index.html [hunk 11]
```diff
@@ -1,74 +1,95 @@
...
+        <button id="experience-button" onclick="moveToExperience()">Experience</button>
+        <button id="education-button" onclick="moveToEducation()">Education</button>
+        <button>Projects</button>
+        <button id="interests-button" onclick="moveToInterest()">Interests</button>
+      </div>
+    </div>
+    <div class="flex-hor">
+      <div>
+        <h1 style="margin-top:50%;"> HI ! I am </h1>
+        <h1 class="prim_color">>Nourish Cherish</h1>
```
Comment (josueetcom) [inline portfolio/src/main/webapp/index.html]: You have an extraneous `>` here:
```suggestion
        <h1 class="prim_color">Nourish Cherish</h1>
```

Surprising the HTML validator did not catch this.

File: portfolio/src/main/webapp/index.html [hunk 12]
```diff
@@ -1,74 +1,95 @@
...
+        <h3>I dont just code. I also teach computers how to compile code</h3>
+      </div>
+      <img id="pfp" src="images/pfp_sps.png" alt="profile picture" />
+    </div>
+    <div class="flex-hor" id="education">
+      <h1 class="prim_color">Education</h1>
+      <div class="card">
+        <h2>Computer Engineering Co-op (2019-2024)</h2>
+        <p>I am a third year Computer Engineering Coop student at the University of Waterloo.
+          Taken courses on Introduction to programming, Data structure and algorithms,Sytems programming and
```
Comment (josueetcom) [inline portfolio/src/main/webapp/index.html]:
```suggestion
          Taken courses on Introduction to programming, Data structure and algorithms, Systems programming and
```

File: portfolio/src/main/webapp/index.html [hunk 13]
```diff
@@ -1,74 +1,95 @@
...
+          concurrency, Real time operating systems development and Computer architectures</p>
+      </div>
+    </div>
+    <div class="flex-hor" id="about">
+      <h1 style="margin:auto" class="prim_color">About</h1>
+      <div class="card">
+        <h2>Me</h2>
+        <p>I enjoy learning about what happening under the hood of operating systems.
```
Comment (josueetcom) [inline portfolio/src/main/webapp/index.html]:
```suggestion
        <p>I enjoy learning about what happens under the hood of operating systems.
```

Comment (josueetcom) [reply portfolio/src/main/webapp/index.html]: OS was my favorite course in school too btw!

**Review by josueetcom — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html (hunk 11 above)
Comment (nourishnew) [reply portfolio/src/main/webapp/index.html]: I guess i found a bug in the html validator. haha

File: portfolio/src/main/webapp/style.css [hunk 14]
```diff
@@ -1,41 +1,46 @@
...
 
-#nav-bar {
+#nav-bar{
   display: flex;
   flex-direction: row;
 }
-.prim_color{
-  color:#fca61f;
+
+.prim-color{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:10]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 15]
```diff
@@ -1,41 +1,46 @@
...
 
-.flex-hor {
+.flex-hor{
   display: flex;
   flex-direction: row;
   justify-content: space-between;
 }
-.flex-col {
+
+.flex-col{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:24]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 16]
```diff
@@ -1,41 +1,46 @@
...
   display: flex;
   flex-direction: column;
 }
+
 form{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:29]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 17]
```diff
@@ -1,41 +1,46 @@
...
-  margin:auto;
+  margin: auto;
   max-width: 70em;
 }
+
 textarea{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:34]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 18]
```diff
@@ -1,41 +1,46 @@
...
-  padding:10px;
+  resize: none;
+  padding: 10px;
   font-family: 'Nunito Sans', sans-serif;
-  margin:10px;
+  margin: 10px;
 }
-#form-submit {
+
+#form-submit{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:43]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 19]
```diff
@@ -58,35 +63,34 @@ button {
   padding: 15px;
 }
 
-
-button:hover {
+button:hover{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:66]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 20]
```diff
@@ -97,26 +101,31 @@ h3 {
   width: 50%;
 }
 
-p {
+p{
   color: #242d49;
   font-family: 'Montserrat', sans-serif;
 }
+
 .font-style{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:109]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 21]
```diff
@@ -97,26 +101,31 @@ h3 {
...
   font-family: 'Montserrat', sans-serif;
 }
+
 .box-shadow{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:113]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 22]
```diff
@@ -97,26 +101,31 @@ h3 {
...
   border-radius: 10px;
   box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
 }
+
 .margin-div{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:118]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 23]
```diff
@@ -97,26 +101,31 @@ h3 {
...
   padding: 5em;
   margin: 5em;
 }
+
 .margin-left{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:123]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 24]
```diff
@@ -97,26 +101,31 @@ h3 {
...
-  margin-left:10px;
+  margin-left: 10px;
 }
+
 .margin-top{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:127]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 25]
```diff
@@ -1,23 +1,59 @@
...
   display: flex;
   flex-direction: row;
 }
 
-#scroll-buttons {
+.prim-color{
+  color: #fca61f;
+}
+
+#scroll-buttons{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:14]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 26]
```diff
@@ -1,23 +1,59 @@
...
+  background-color: #fca61f;
+  border-radius: 25px;
+  color: antiquewhite;
+  font: bold medium/1.2 Arial, sans-serif;
+  padding: 15px;
+  cursor: pointer;
+  max-width: 25em;
+}
+
+button{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:56]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

File: portfolio/src/main/webapp/style.css [hunk 27]
```diff
@@ -27,35 +63,34 @@ button {
...
-h3 {
+h3{
   color:#242d49;
   font-family: 'Indie Flower', cursive;
   font-size: small;
-
 }
 
-#experience-view {
+#experience-view{
```
Comment (josueetcom) [inline portfolio/src/main/webapp/style.css:93]: [4.2.5](https://google.github.io/styleguide/htmlcssguide.html#Property_Name_Stops): Use a space between the last selector and the declaration block.

**Review by josueetcom — APPROVED** [review]

**Pull request merged**